nautobot/core/ui/object_detail.py

```diff
@@ -922,12 +1033,13 @@ def __init__(
     def render_label(self, context: Context):
         """Default to rendering the provided object's `verbose_name` if no more specific `label` was defined."""
         if self.label is None:
-            return bettertitle(get_obj_from_context(context)._meta.verbose_name)
+            return bettertitle(get_obj_from_context(context, self.context_object_key)._meta.verbose_name)
         return super().render_label(context)
 
     def render_value(self, key, value, context: Context):
+        obj = get_obj_from_context(context, self.context_object_key)
         try:
-            field_instance = get_obj_from_context(context)._meta.get_field(key)
+            field_instance = obj._meta.get_field(key)
```

Comment:
nitpick:
Should we set `self.obj = get_obj_from_context(context, self.context_object_key)` so we can reference it in these functions?

Reply:
Potentially, though we'd have to define which function gets called "first" to do this assignment. Can't do it in `__init__()` since we don't have context at that point.